freedomofpress/securedrop.org
# Delisted state

Add a `delisted` field to directory entries that allows an entry to be marked as delisted with a preset reason. Delisted entries do not appear on the directory page nor in the API, but their page can remain live and renders with a warning:

Resolves #491

![screen shot 2018-08-22 at 4 45 08 pm](https://user-images.githubusercontent.com/296631/44491344-f261c400-a62e-11e8-87c5-549ab85aadc5.png)

### Testing

On this branch:

1. run migrations
2. edit a directory entry's delisted status to something other than the default (this field appears under the "Settings" tab)
3. observe that the entry does not render in the directory
4. observe that the entry does not render in the API
5. observe that you can visit the entry's page but it renders without the landing page URL or onion service but with a warning instead
